src/impy/models/dpmjetIII.py

```diff
@@ -81,6 +82,10 @@ def __init__(self, event_kinematics, seed=None, logfname=None):
         dpm_conf = impy_config["dpmjetIII"]
 
         info(1, "Initializing DPMJET-III")
+
+        _check_model_data_files(
+            pathlib.Path(dpm_conf["dat_dir"][self.version]).parts[-1]
+        )
         # Set the dpmjpar.dat file
```

Comment:
This is what I wanted to avoid by encapsulating all the file retrival in CachedPath that derives from Path. For most or all models, we need to construct a Path objects anyway to initialize the model. Instead of a Path, one makes a CachedPath and the rest works automatically.

Reply:
As I understand models needs paths as strings. Do we need Path only for path construction?

Reply:
Indeed, Path objects are not automatically converted to string when the interface requires it. Whether that is an oversight or intended to make the conversion explicit (which is a good thing), I don't know, but you simply do something like this 
```py
datapath = CachedPath("localpath", "remotepath")
self._lib.init_my_model(str(datapath))
```

Reply:
Of course, if we never "work" with the CachedPath as an object (use the / operator, check for existence, etc), then you are right that this can be simplified similar to your code to
```
local_path = _cached_path("remotepath")
```
where `_cached_path` is just a function that returns the local path as a string. The local path can be inferred from the remote path. We need some kind of convention anyway on how these paths and files are named. If we need a convention anyway, we can bake it into code.